feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast-event.listener.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  ADDED_TO_CART_FEEDBACK,
+  CartConfig,
+  CartUiEventAddToCart,
+} from '@spartacus/cart/base/root';
+import { CxEvent, EventService } from '@spartacus/core';
+import { LaunchDialogService } from '@spartacus/storefront';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import { AddedToCartToastEventListener } from './added-to-cart-toast-event.listener';
+
+const mockEventStream$ = new BehaviorSubject<CxEvent>({});
+
+class MockEventService implements Partial<EventService> {
+  get(): Observable<any> {
+    return mockEventStream$.asObservable();
+  }
+}
+
+const mockEvent = new CartUiEventAddToCart();
+mockEvent.productCode = 'test';
+mockEvent.quantity = 3;
+mockEvent.numberOfEntriesBeforeAdd = 1;
+
+const mockInstance = {
+  init: () => {},
+  addToast: (_event: CartUiEventAddToCart) => {},
+};
+
+const mockToastRef = { instance: mockInstance };
+
+class MockLaunchDialogService implements Partial<LaunchDialogService> {
+  launch(): Observable<any> {
+    return of(mockToastRef);
+  }
+}
+
+const mockCartConfig: CartConfig = {
+  cart: {
+    addToCartFeedback: {
+      feedback: ADDED_TO_CART_FEEDBACK.TOAST,
+      toast: { timeout: 3000 },
+    },
+  },
+};
+
+describe('AddedToCartToastEventListener', () => {
+  let listener: AddedToCartToastEventListener;
+  let launchDialogService: LaunchDialogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        AddedToCartToastEventListener,
+        { provide: EventService, useClass: MockEventService },
+        {
+          provide: LaunchDialogService,
+          useClass: MockLaunchDialogService,
+        },
+        {
+          provide: CartConfig,
+          useValue: mockCartConfig,
+        },
+      ],
+    });
+
+    listener = TestBed.inject(AddedToCartToastEventListener);
+    launchDialogService = TestBed.inject(LaunchDialogService);
+  });
+
+  describe('onAddToCart', () => {
```

Comment:
We should also test that the toast is not displayed if the config is not set to display the toast